apps/api/src/controllers/rewards.controller.ts

```diff
@@ -0,0 +1,131 @@
+import { NextFunction, Request, Response } from "express";
+
+import { serviceLogger } from "@/lib/logger.js";
+import { ApiError } from "@/middleware/errorHandler.js";
+import { ServiceRegistry } from "@/services/index.js";
+
+// No query parameters needed - wallet address comes from authenticated session
+
+/**
+ * Rewards Controller
+ * Handles reward-related operations with SIWE session authentication
+ * All endpoints require authenticated user session (req.userId and req.wallet)
+ */
+export function makeRewardsController(services: ServiceRegistry) {
+  return {
+    /**
+     * Get total claimable rewards for the authenticated user
+     * GET /api/user/rewards/total
+     * @param req Express request with wallet from session
+     * @param res Express response
+     * @param next Express next function
+     */
+    async getTotalClaimableRewards(
+      req: Request,
+      res: Response,
+      next: NextFunction,
+    ) {
+      try {
+        const address = req.wallet as string;
```

Comment:
We should use Zod for this instead of casting.  That will catch problems that arise from the string "undefined" being passed into functions.

There's two other places in this file that have the same issue, line 68 and 103.